src/Common/Core/Test/Fixtures/CoreShellProviderFixture.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.ComponentModel.Composition.Hosting;
+using System.Threading.Tasks;
+using Microsoft.Common.Core.Extensions;
+using Microsoft.Common.Core.Services;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Common.Core.Test.Fakes.Shell;
+using Microsoft.UnitTests.Core.Shell;
+using Microsoft.UnitTests.Core.XUnit;
+using Microsoft.UnitTests.Core.XUnit.MethodFixtures;
+using Xunit.Sdk;
+
+namespace Microsoft.Common.Core.Test.Fixtures {
+    public abstract class CoreShellProviderFixture : MethodFixtureBase, ICoreShellProvider {
```

Comment:
I think we can move remaining fields to some `IApplicationService` and then only raw 'IServiceContainer` will remain and then we won't need `ICoreShell` at all. This will take lots of renames though so perhaps next time around.